cpp/src/lance/encodings/plain.h

```diff
@@ -68,6 +68,28 @@ class PlainDecoder : public Decoder {
     return std::make_shared<ArrayType>(length.value(), buf);
   }
 
+  ::arrow::Result<std::shared_ptr<::arrow::Array>> Take(
```

Comment:
Dumb question: if we want say OFFSET 1000 LIMIT 10000, is this the method that gets called? If yes, does this mean the `indices` have to contain all of range(1000,10000)?

Reply:
There will be two ways to "get an array from decoder", which will be implemented in #32 eventually.

So first, is scan without filter, that, will just call `Decoder::ToArray(start, length)` for performance.

Bu with filters, the logic will essentially be:

```c++
Array mask;
for (auto filtered_column :  all_filter_columns) {
   mask &= filtered_column;  // or `mask |= filtered_column` depends on the expression
}
auto indices = MaskToIndices(mask);
for (auto projected_field : all_projected_field) {
   project_field->GetDecoder()->Take(indices)
}
```

Reply:
Oh, also, the `indices` are within a "chunk" / "page".